### PR TITLE
validateIndentation: exception to indentation rules for module pattern

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -147,6 +147,33 @@ exports.isSnakeCased = function(name) {
 };
 
 /**
+ * Returns the function expression node if the provided node is an iffe,
+ * other returns undefined.
+ *
+ * @param  {Object} node
+ * @return {Object|undefined}
+ */
+exports.getFunctionNodeFromIIFE = function(node) {
```
Comment (markelog) [inline lib/utils.js]: This method could be used anywhere else?

Comment (markelog) [inline lib/utils.js]: Strike that, see the use

File: lib/utils.js [hunk 2]
```diff
@@ -147,6 +147,33 @@ exports.isSnakeCased = function(name) {
 };
 
 /**
+ * Returns the function expression node if the provided node is an iffe,
+ * other returns undefined.
+ *
+ * @param  {Object} node
+ * @return {Object|undefined}
```
Comment (markelog) [inline lib/utils.js]: Usually, if there is no object to return, null is used instead of undefined.

File: test/rules/validate-indentation.js [hunk 3]
```diff
@@ -150,6 +150,67 @@ describe('rules/validate-indentation', function() {
...
+        beforeEach(function() {
+            checker.configure({ validateIndentation: 4 });
+        });
+
+        var cases = {
+            'no indentation': 'a++;',
+            indentation: '    a++;'
+        };
+
+        Object.keys(cases).forEach(function(title) {
```
Comment (markelog) [inline test/rules/validate-indentation.js]: These kind of test declarations are harder to isolate, not sure if copy/paste is worse, i did it couple times myself though

**Comment by markelog** [comment]
I would divide this commit, `require-parentheses-around-iife` changes are nothing to do with #436.

Code looks better, comments especially helpful, but i'm afraid bus factor of the validateIndentation rule is equal to one. @zxqfox, @mdevils, @mrjoelkemp i'm i right?

**Comment by mikesherov** [comment]
Yes, hit by a bus factor is still high. 

I recently spent a lot of time cleaning up this rule to reduce the complexity of the algorithm. 

I also spent a bit of time with @mrjoelkemp reviewing the rule and how it works.

Next, I'm going to write up how this rule works, because the plan is to have other indentation rules extend from it and use its interface (which I have not exposed yet).

The hope is to have someone else write one of the other indentation rules using the interface. That way, when I buy a private island, all knowledge of indentation will not be lost :-)

File: lib/utils.js (hunk 2 above)
Comment (mikesherov) [inline lib/utils.js]: Good point. I'll change that.

Comment (zxqfox) [inline lib/utils.js]: `{?Object}` ?

File: test/rules/validate-indentation.js (hunk 3 above)
Comment (mikesherov) [inline test/rules/validate-indentation.js]: I think copy/paste is worse in this case. I'm going to leave it unless you have strong objection.

File: lib/utils.js (hunk 2 above)
Comment (mikesherov) [inline lib/utils.js]: @zxqfox, if that's valid JSDoc, I'll use it.

Comment (zxqfox) [inline lib/utils.js]: @mikesherov Absolutely valid. http://usejsdoc.org/tags-type.html It's called nullable type.

Comment (mrjoelkemp) [inline lib/utils.js]: @zxqfox That's cool. Great tip.

**Comment by mikesherov** [comment]
Thanks for the review. Landed with the proposed modifications!

**Pull request closed without merging**